From db5a4a9ae7b0048f033361744421da8569642f73 Mon Sep 17 00:00:00 2001
From: Werner Lemberg <wl@gnu.org>
Date: Sat, 12 Sep 2015 08:32:55 +0200
Subject: [psaux] Fix potential buffer overflow (#45922).

* src/psaux/psobjs.c (ps_parser_skip_PS_token): If a token is
enclosed in balanced expressions, ensure that the cursor position
doesn't get larger than the current limit.
---
 src/psaux/psobjs.c | 3 +++
 1 file changed, 3 insertions(+)

(limited to 'src/psaux/psobjs.c')

diff --git a/src/psaux/psobjs.c b/src/psaux/psobjs.c
index c7cbc67c6..54eabe2b3 100644
--- a/src/psaux/psobjs.c
+++ b/src/psaux/psobjs.c
@@ -594,6 +594,9 @@
       error = FT_THROW( Invalid_File_Format );
     }
 
+    if ( cur > limit )
+      cur = limit;
+
     parser->error  = error;
     parser->cursor = cur;
   }
-- 
cgit v1.2.1

